### PR TITLE
Create xiaomi_gw.markdown

### PR DESCRIPTION
**Pull request by jcastro**
Initial commit for the new Xiaomi Gateway component

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

### CODE REVIEW

File: source/_components/xiaomi_gw.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+{% linkable_title Getting started %}
+
+Follow the setup process using your phone and Mi Home app. From here you will be able to retrieve the key from within the app following [this tutorial](https://community.home-assistant.io/t/beta-xiaomi-gateway-integration/8213/1832)
+
+{% linkable_title Configuration %}
+
+One Gateway
+  ```yaml
+ # You can leave sid empty if you only have one gateway
```
Comment (Danielhiversen) [inline source/_components/xiaomi_gw.markdown]: I have changed from sid to mac address in the implementation now.
So the config should be:` - mac: `
Both the old sid format and mac address with colon are supported

File: source/_components/xiaomi_gw.markdown [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+title: "Xiaomi Gateway”
+description: "Instructions how to integrate your Xiaomi Gateway within Home Assistant."
+date: 2017-07-21 16:34
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: xiaomi.png
+ha_category: Hub
+ha_release: 0.4X
```
Comment (Danielhiversen) [inline source/_components/xiaomi_gw.markdown]: It will be 0.50

File: source/_components/xiaomi_gw.markdown (hunk 1 above)
Comment (jcastro) [inline source/_components/xiaomi_gw.markdown]: changed 👍 thanks

Comment (Danielhiversen) [inline source/_components/xiaomi_gw.markdown]: Great (replace sid on line 55 with mac too)

File: source/_components/xiaomi_gw.markdown [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   gateways:
+     - mac: xxxxxxxxxxxx
+       key: xxxxxxxxxxxxxxxx
+     - mac: xxxxxxxxxxxx
+       key: xxxxxxxxxxxxxxxx
+  ```
+
+Configuration variables:
+- **mac** (*Optional*): The MAC of your gateway. Required if you have more than one gateway
+- **key** (*Required*): The key of your gateway
```
Comment (Danielhiversen) [inline source/_components/xiaomi_gw.markdown]: key is optional, but without providing the key you can not control lights or the switch. (But the sensors and binary sensors will still work)
**discovery_retry**  (*Optional*): How many times Home Assistant should try to reconnect to the Xiaomi hub. (Default is 3.)
**interface**  (*Optional*): Which network interface to use. (Default to `any`)

File: source/_components/xiaomi_gw.markdown [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+       key: xxxxxxxxxxxxxxxx
+     - mac: xxxxxxxxxxxx
+       key: xxxxxxxxxxxxxxxx
+  ```
+
+Configuration variables:
+- **mac** (*Optional*): The MAC of your gateway. Required if you have more than one.
+- **key** (*Optional*): The key of your gateway. Required if you also want to control lights and switches; sensors and binary sensors will still work.
+- **discovery_retry** (*Optional*): Amount of times Home Assitant should try to reconnect to the Xiaomi Gateway. Default is 3.
+- **interface** (*Optional*): Which network interface to use. Default to any.
```
Comment (syssi) [inline source/_components/xiaomi_gw.markdown]: {% linkable_title Services %}

The gateway provides two services (xiaomi.play_ringtone, xiaomi.stop_ringtone). To play ringtones by HA the version of the gateway firmware must be 1.4.1_145 at least. A `ringtone_id` and `gw_mac` must be supplied. The parameter 'ringtone_vol' (percent) is optional. Allowed values of the `ringtone_id` are:

- alarm ringtones [0-8]
- doorbell ring [10-13]
- alarm clock [20-29]
- custom ringtones (uploaded by mi home app) starting from 10001

Automation example

```yaml
    - alias: Let a dog bark on long press
      trigger:
        platform: event
        event_type: click
        event_data:
          entity_id: binary_sensor.switch_158d000xxxxxc2
          click_type: long_click_press
      action:
        service: xiaomi.play_ringtone
        data:
          gw_mac: xxxxxxxxxxxx
          ringtone_id: 8
          ringtone_vol: 8

    - alias: Stop barking immediately on single click
      trigger:
        platform: event
        event_type: click
        event_data:
          entity_id: binary_sensor.switch_158d000xxxxxc2
          click_type: single
      action:
        service: xiaomi.stop_ringtone
        data:
          gw_mac: xxxxxxxxxxxx
```

**Comment by EabeyTong** [comment]
I hope that the mi gateway can join the control play FM

**Pull request merged**